internal/services/firewall/firewall_policy_application_rule_collection_resource.go

```diff
@@ -0,0 +1,432 @@
+
+	rules, err := flattenFirewallPolicyRuleApplication(ruleCollection.Rules)
+	if err != nil {
+		return fmt.Errorf("flattening Firewall Policy Rule Collections: %+v", err)
+	}
+
+	if err := d.Set("rule", rules); err != nil {
+		return fmt.Errorf("setting `rules`: %+v", err)
+	}
+
+	return nil
+}
+
+func resourceFirewallPolicyApplicationRuleCollectionUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Firewall.FirewallPolicyRuleGroupClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.FirewallPolicyRuleCollectionID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.FirewallPolicyName, id.RuleCollectionGroupName)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[DEBUG] Firewall Policy Rule Collection Group %q was not found in Resource Group %q - removing from state!", id.RuleCollectionGroupName, id.ResourceGroup)
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("retrieving Firewall Policy Rule Collection Group %q (Resource Group %q / Policy: %q): %+v", id.RuleCollectionGroupName, id.ResourceGroup, id.FirewallPolicyName, err)
+	}
+
+	idx := indexFunc(*resp.RuleCollections, func(policy network.BasicFirewallPolicyRuleCollection) bool {
+		info, _ := policy.AsFirewallPolicyFilterRuleCollection()
+		return *info.Name == id.RuleCollectionName
+	})
+
+	if idx == -1 {
+		log.Printf("[DEBUG] Firewall Policy Rule Collection %q was not found in Firewall Policy Rule Collection Group %q - removing from state!", id.RuleCollectionName, id.RuleCollectionGroupName)
+		d.SetId("")
+		return nil
+	}
+
+	locks.ByName(id.FirewallPolicyName, AzureFirewallPolicyResourceName)
+	defer locks.UnlockByName(id.FirewallPolicyName, AzureFirewallPolicyResourceName)
+
+	rules := expandFirewallPolicyRuleApplication(d.Get("rule").([]interface{}))
+	ruleCollection := &network.FirewallPolicyFilterRuleCollection{
+		Action: &network.FirewallPolicyFilterRuleCollectionAction{
+			Type: network.FirewallPolicyFilterRuleCollectionActionType(d.Get("action").(string)),
+		},
+		Name:               utils.String(d.Get("name").(string)),
+		Priority:           utils.Int32(int32(d.Get("priority").(int))),
+		RuleCollectionType: network.RuleCollectionTypeFirewallPolicyFilterRuleCollection,
+		Rules:              rules,
+	}
+
+	if idx == -1 {
+		ruleCollections := append(*resp.RuleCollections, ruleCollection)
+		resp.RuleCollections = &ruleCollections
+	} else {
+		(*resp.RuleCollections)[idx] = ruleCollection
+	}
+
+	param := network.FirewallPolicyRuleCollectionGroup{
+		FirewallPolicyRuleCollectionGroupProperties: &network.FirewallPolicyRuleCollectionGroupProperties{
+			Priority:        resp.Priority,
+			RuleCollections: resp.RuleCollections,
+		},
+	}
+
```

Comment:
Can we simply re-assign the `RuleCollections` of the `resp` to avoid missing constructing any new properties in the future?